Library/Formula/pre-commit.rb

```diff
@@ -0,0 +1,85 @@
+class PreCommit < Formula
+  homepage "http://pre-commit.com/"
+  url "https://github.com/pre-commit/pre-commit/archive/v0.5.0.tar.gz"
+  sha256 "567d8e0f64e176c2c2c97c80a3c96c2d40c7a21d205bee84ce5f74d9323de9fb"
+
+  depends_on :python if MacOS.version <= :snow_leopard
+
+  resource "argparse" do
+    url "https://pypi.python.org/packages/source/a/argparse/argparse-1.3.0.tar.gz"
+    sha256 "b3a79a23d37b5a02faa550b92cbbbebeb4aa1d77e649c3eb39c19abf5262da04"
+  end
+
+  resource "aspy.yaml" do
+    url "https://pypi.python.org/packages/source/a/aspy.yaml/aspy.yaml-0.2.1.tar.gz"
+    sha256 "a91370183aea63c87d8487e7b399ed2d99a7c2f14b108d27c0bc8ad9ef595d9a"
+  end
+
+  resource "cached-property" do
+    url "https://pypi.python.org/packages/source/c/cached-property/cached-property-1.2.0.tar.gz"
+    sha256 "e3081a8182d3d4b7283eeade76c382bcfd4dfd644ca800598229c2ef798abb53"
+  end
+
+  resource "jsonschema" do
+    url "https://pypi.python.org/packages/source/j/jsonschema/jsonschema-2.4.0.tar.gz"
+    sha256 "1298a2f1b2f4c4a7b921cccd159e4e42f6d7b0fb75c86c0cdecfc71f061833fa"
+  end
+
+  resource "nodeenv" do
+    url "https://pypi.python.org/packages/source/n/nodeenv/nodeenv-0.13.2.tar.gz"
+    sha256 "626881a22f252635fe2bc72b554e1fbb2b80bcd626103de21ed112ea0c024e37"
+  end
+
+  resource "ordereddict" do
+    url "https://pypi.python.org/packages/source/o/ordereddict/ordereddict-1.1.tar.gz"
+    sha256 "1c35b4ac206cef2d24816c89f89cf289dd3d38cf7c449bb3fab7bf6d43f01b1f"
+  end
+
+  resource "pyyaml" do
+    url "https://pypi.python.org/packages/source/P/PyYAML/PyYAML-3.11.tar.gz"
+    sha256 "c36c938a872e5ff494938b33b14aaa156cb439ec67548fcab3535bb78b0846e8"
+  end
+
+  resource "simplejson" do
+    url "https://pypi.python.org/packages/source/s/simplejson/simplejson-3.7.1.tar.gz"
+    sha256 "4b837e37885503776b8d76182aa91e3bb27a3c38a023b14008e4cb63170bebc4"
+  end
+
+  resource "virtualenv" do
+    url "https://pypi.python.org/packages/source/v/virtualenv/virtualenv-12.1.1.tar.gz"
+    sha256 "3c88cf7df114c32cf0651d12f979b8db556d992d647493e5bdbbe2828be40007"
+  end
+
+  def install
+    ENV["PYTHONPATH"] = libexec/"vendor/lib/python2.7/site-packages"
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+
+    resources.each do |r|
+      r.stage do
+        system "python", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    # fix aspy.yaml
```

Comment:
I had a quick note on the commit with the fix about this. `aspy.yaml` uses a `.pth` file that's normally parsed automatically during python's initialization but only if it's in one of the default `site-packages` locations, which is not the case with vendor'ed resources. The fix just declares the `aspy` dir as a module, which is what the `.pth` file programatically does.

Do python homebrew formulas have a standard way of dealing with vendor dependencies that use `.pth` files to declare libraries?

(More info in the first 3 paragraphs: https://docs.python.org/2/library/site.html)